### PR TITLE
Add reset_message_labels.py

### PR DESCRIPTION
**Pull request by as2388**
This makes it a bit easier to fix cycles in Coda while correctly preserving history in the database

### CODE REVIEW

File: engagement_database/reset_message_labels.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import argparse
+import json
+import subprocess
+
+from core_data_modules.cleaners import Codes, URNCleaner
```
Comment (DanielMwendwa) [inline engagement_database/reset_message_labels.py:5]: Unused imports

File: engagement_database/reset_message_labels.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+from core_data_modules.logging import Logger
+from engagement_database import EngagementDatabase
+from engagement_database.data_models import HistoryEntryOrigin, CommandLogEntry, CommandStatuses
+from google.cloud import firestore
+from id_infrastructure.firestore_uuid_table import FirestoreUuidTable
```
Comment (DanielMwendwa) [inline engagement_database/reset_message_labels.py:10]: Unused imports

File: engagement_database/reset_message_labels.py [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+from storage.google_cloud import google_cloud_utils
+
+log = Logger(__name__)
+
+BATCH_SIZE = 500
```
Comment (DanielMwendwa) [inline engagement_database/reset_message_labels.py:15]: Unused variable

File: engagement_database/reset_message_labels.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+    parser = argparse.ArgumentParser(description="Resets all of a message's labels and previous_datasets, and returns "
+                                                 "the message back to its original dataset")
+
+    parser.add_argument("--dry-run", const=True, default=False, action="store_const")
+    parser.add_argument("user", help="Identifier of the user launching this program")
+    parser.add_argument("google_cloud_credentials_file_path", metavar="google-cloud-credentials-file-path",
+                        help="Path to a Google Cloud service account credentials file to use to access the "
+                             "credentials bucket")
+    parser.add_argument("engagement_database_credentials_file_url", metavar="engagement-database-credentials-file-url",
+                        help="GS URL of the credentials for the Firestore project to export")
```
Comment (DanielMwendwa) [inline engagement_database/reset_message_labels.py:46]: `help="GS URL to the Engagement firestore database credentials file")`

File: engagement_database/reset_message_labels.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    parser.add_argument("database_path", metavar="database-path",
+                        help="Path to the engagement database to export e.g. engagement_databases/test")
```
Comment (DanielMwendwa) [inline engagement_database/reset_message_labels.py:48]: `...engagement database to update` not export

**Review by DanielMwendwa — APPROVED** [review]

File: engagement_database/reset_message_labels.py (hunk 1 above)
Comment (as2388) [reply engagement_database/reset_message_labels.py]: optimised imports

File: engagement_database/reset_message_labels.py (hunk 3 above)
Comment (as2388) [reply engagement_database/reset_message_labels.py]: removed

File: engagement_database/reset_message_labels.py (hunk 4 above)
Comment (as2388) [reply engagement_database/reset_message_labels.py]: updated

File: engagement_database/reset_message_labels.py (hunk 5 above)
Comment (as2388) [reply engagement_database/reset_message_labels.py]: updated

**Pull request merged**